integration-tests/test-app/index.ts

```diff
@@ -32,6 +34,9 @@ export const app = () => {
 
 	return html`
 		<main>
+			<!-- for regression testing, it is important that document-title-setter is the first element -->
+			<document-title-setter />
```

Comment:
Why is it important that the document-title-setter is the first element? What would happen if you needed some sort of error boundary before this? Or is it static?

Reply:
as long as it is the first resolved element (specifically the first element to have its effects resolved, which depends on it be the first leaf-node), then the tests will work as expected - the issue is if this was ever moved, the tests would erroneously pass.